challengeutils/wiki.py

```diff
@@ -0,0 +1,208 @@
+"""Pull and push wiki tools"""
+import json
+import os
+import typing
+
+import synapseclient
+from synapseclient import Synapse
+
+
+def pull_wiki(syn: Synapse, project: str,
+              workdir: str = "./") -> typing.List[dict]:
+    """Downloads each wikipage's content into a markdown file and
+    stores a configuration file
+
+    Args:
+        syn: Synapse connection
+        project: synapseclient.Project or its id
+        workdir: Location to download markdown files and wiki_config.json
+                 into. Defaults to location of where code is being
+                 executed.
+
+    Returns:
+        Wiki Configuration
+        [
+            {
+                "id": "111",
+                "title": "title",
+                "parentId": "33333",
+                "markdown_path": "home.md"
+            },
+            {...}
+        ]
+
+    """
+    projectid = synapseclient.core.utils.id_of(project)
+    wiki_headers = syn.getWikiHeaders(projectid)
+    for wiki_header in wiki_headers:
+        wiki = syn.getWiki(projectid, subpageId=wiki_header['id'])
+        # Convert all special characters to underscore
+        title = ''.join(letter for letter in wiki['title']
```

Comment:
Just wanted to confirm whether this replaces special characters with underscores (as noted by the comment)?

I'm interpreting this line as not including the special characters in `title`.

Reply:
Hmm @vpchung , this variable is poorly named.  Im simply removing the special characters so that there are none in the markdown file path (Makes it easier to operate when in shelll)

Reply:
Ah okay, sounds good!  The comment above just says that the special characters will be replaced with underscores, but I didn't see where the operation was doing that.  Like you said, it was just removing them haha.